### ISSUE TITLE
LogDebug and LogTrace with the category only fail to compile

### ISSUE THREAD

**Issue by makortel** [cms-sw/cmssw#32285]
Code like
```
LogTrace("only_category");
LogDebug("only_category");
```
fails to build (in CMSSW_11_2_X_2020-11-24-1100) when `EDM_ML_DEBUG` is defined. The compiler error messages are
```
.../CMSSW_11_2_X_2020-11-24-1100/src/FWCore/MessageLogger/interface/MessageLogger.h:229:45: error: cannot bind non-const lvalue reference of type 'edm::Log<edm::level::Debug, false>&' to an rvalue of type 'edm::Log<edm::level::Debug, false>'
       : edm::impl::LogDebugAdapter() | edm::LogDebug_(id, __FILE__, __LINE__)
                                             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
.../CMSSW_11_2_X_2020-11-24-1100/src/FWCore/MessageService/test/UnitTestClient_Hd.cc:20:5: note: in expansion of macro 'LogDebug'
     LogDebug("only_category");
     ^~~~~~~~
..../CMSSW_11_2_X_2020-11-24-1100/src/FWCore/MessageLogger/interface/MessageLogger.h:162:17: note:   initializing argument 1 of 'edm::LogDebug_ edm::impl::LogDebugAdapter::operator|(edm::Log<edm::level::Debug, false>&)'
       LogDebug_ operator|(Log<level::Debug, false>& iOther) { return LogDebug_(iOther); }
                 ^~~~~~~~

```

This problem was noticed in static analyzer errors in https://github.com/cms-sw/cmssw/pull/32093#issuecomment-733719188, and is actually also visible in static analyzer failures (not reports) in the IBs.

**Comment by makortel** [comment]
assign core

**Comment by makortel** [comment]
The fix would be straightforward, but I started to wonder how useful category-only `LogDebug` and `LogTrace` would really be. Such `LogDebug` would result in a message with the category name, but `LogTrace`would only produce an empty line. I would also imagine the intention of such code to really display such a *message*  instead of a *category* (i.e. such uses would be mostly incorrect uses of MessageLogger).

On the other hand, if we decide to disallow such uses for 'LogDebug` and `LogTrace`, I think it would be good to do the same for other message severity levels as well for consistency (but I don't know how to do that without runtime check, which I'd really like to avoid).

**Comment by makortel** [comment]
Fixed in #33106

**Comment by makortel** [comment]
+1
